File: Striver/Trees/Q10.py
# maximum depth of binary tree

from Node import Node

def maxDepth(root):
    if not root:
        return 0
    return 1 + max(maxDepth(root.left), maxDepth(root.right))

root = Node(1)
root.left = Node(2)
root.right = Node(3)
root.left.left = Node(4)
root.left.left.left = Node(8)
root.left.left.right = Node(9)
root.left.right = Node(5)
root.right.left = Node(6)
root.right.right = Node(7)

print(maxDepth(root))

File: Striver/Trees/Q10.java
package Striver.Trees;

public class Q10{
    // Maximum Depth of Binary Tree

    public int maxDepth(Node root) {
        if(root == null) return 0;
        return 1 + Math.max(maxDepth(root.left),maxDepth(root.right));
    }

    public static void main(String[] args) {
        Node root = new Node(3);
        root.left = new Node(9); root.right = new Node(20);
        root.right.left = new Node(15); root.right.right = new Node(7);
        Q10 sol = new Q10();
        
        System.out.println(sol.maxDepth(root));
    }

}